### PR TITLE
fix(portal): Fix all website links in product

### PR DESCRIPTION
**Pull request by jamilbk**
Adds a dedicated `anchor` attr to the `website_link` component because the order of components in a URL must be `/path?params#anchor`.

### CODE REVIEW

File: elixir/apps/web/lib/web/components/navigation_components.ex [hunk 1]
```diff
@@ -331,13 +331,14 @@ defmodule Web.NavigationComponents do
...
   attr :href, :string, required: true
+  attr :anchor, :string, required: false, default: ""
   slot :inner_block, required: true
   attr :rest, :global
 
   def website_link(assigns) do
     ~H"""
     <.link
-      navigate={"https://www.firezone.dev?utm_source=product#{@href}"}
+      navigate={"https://www.firezone.dev#{@href}?utm_source=product##{@anchor}"}
```
Comment (bmanifold) [inline elixir/apps/web/lib/web/components/navigation_components.ex]: Should we rename `href` to `path`?  An actual `href` can take a full URL, but in this case that would not work.

Comment (ReactorScram) [reply elixir/apps/web/lib/web/components/navigation_components.ex]: Agreed, then if we change `anchor` to `fragment` it's consistent with the URL parts entirely https://en.wikipedia.org/wiki/URL#Syntax

**Review by ReactorScram — DISMISSED** [review]
Looks good except for the thing Brian pointed out

**Review by AndrewDryga — APPROVED** [review]

File: elixir/apps/web/lib/web/components/navigation_components.ex (hunk 1 above)
Comment (jamilbk) [reply elixir/apps/web/lib/web/components/navigation_components.ex]: SGTM

**Review by ReactorScram — APPROVED** [review]

**Pull request merged**